subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/NestedBeanContext.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.properties;
+
+public interface NestedBeanContext<T extends BeanNode<T>> {
+    T createNode(String propertyName, Object nested);
```

Comment:
There is no need for this method. It's only doing `new T()` at this point, and at every call site we know the value of the `T` parameter. We could simply inline the method instead.

Reply:
Well spotted! Removed.